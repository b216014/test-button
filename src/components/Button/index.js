import React from 'react';
import * as styles from '../index.css';

const Button = ({children, testID, onClick, buttonType,}) => (
  <button className={buttonType === 'rect' ? styles.rect : styles.round} data-testid={testID} onClick={() => onClick()}>
    {children}
  </button>
);

export default Button;
