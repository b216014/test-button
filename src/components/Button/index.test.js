import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import  Button  from './index';
import '../index.css';


describe('the button component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button testID="test-btn">Test text</Button>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call the onClick prop when the button is clicked', () => {
    const onClickMock = jest.fn();
    const { getByTestId } = render(<Button testID="test-btn" onClick={onClickMock}>Test text</Button>);

    fireEvent.click(getByTestId('test-btn'));
    expect(onClickMock).toHaveBeenCalledWith();
  });

  it('should verify the type of button passed (rect) to the component', () => {
    const onClickMock = jest.fn();
    const { buttonType } = render(<Button testID="test-btn" buttonType="rect" onClick={onClickMock} />);

    expect(buttonType).toMatchSnapshot();
  });

});
