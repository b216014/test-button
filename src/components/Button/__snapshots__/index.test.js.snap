// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component should render correctly 1`] = `
<DocumentFragment>
  <button
    data-testid="test-btn"
  >
    Test text
  </button>
</DocumentFragment>
`;

exports[`the button component should verify the type of button passed (rect) to the component 1`] = `undefined`;

exports[`the button component should verify the type of button passed (round) to the component 1`] = `undefined`;

exports[`the button component should verify the type of button passed to the component 1`] = `undefined`;
